src/containers/AccountConnection.jsx

```diff
@@ -86,14 +92,18 @@ class AccountConnection extends Component {
             this.setState({account: account})
             account.folderPath = account.folderPath || t('account.config.default_folder', konnector)
             return this.runConnection(accounts[currentIdx], account.folderPath)
-              .then(() => {
+              .then(connection => {
                 this.setState({
                   connector: konnector,
                   isConnected: konnector.accounts.length !== 0,
                   selectedAccount: currentIdx,
                   submitting: false
                 })
-                this.handleCreateSuccess()
+                if (connection.isSuccessTimedOut) {
```

Comment:
You're right, now I can use a common method to handle these two connection successes 👍